Source/relay/TourGuide/Items of the Month/2020/Cargo Cultist Shorts.ash

```diff
@@ -5,25 +5,31 @@ void IOTMCargoCultistShortsGenerateResource(ChecklistEntry [int] resource_entrie
  
 	if (!get_property_boolean("_cargoPocketEmptied")) {
         string image_name = "__item cargo cultist shorts";
+
+        boolean [int] empty_pockets;
+        string [int] empty_pocket_list = split_string(get_property("cargoPocketsEmptied"), ",");
+        foreach pocket in split_string(get_property("cargoPocketsEmptied"), ",") {
+            empty_pockets[to_int(empty_pocket_list[i])] = true;
+        }
  
 		string [int] options;
 		string [int] description;
         description.listAppend("Pick a pocket for something useful! Too many to list!");
         if (__misc_state["in run"] && my_path().id != PATH_COMMUNITY_SERVICE)
 		{
-        	if (locationAvailable($location[The royal guard Chamber]) == true)
+        	if (locationAvailable($location[The royal guard Chamber]) == true && !(empty_pockets contains 343))
```

Comment:
i am assuming you've checked this tweak operates correctly? part of me thinks that given how you factored this, instead of doing a contains logic like this, you should just do `&& !empty_pockets[343]` because it's slightly more fitting. i *think* the syntax you have here works but that feels slightly cleaner to me.

Reply:
Yes, I checked that this works.